### PR TITLE
feat: #8226 split payments apps

### PR DESCRIPTION
**Pull request by willmcvay**
# Pull request checklist

**Detail as per issue below (required):**

fixes: #8226

### CODE REVIEW

File: packages/payments-client/src/tests/__mocks__/payment.ts [hunk 1]
```diff
@@ -54,23 +53,22 @@ export const mockPaymentModelPagedResult = {
...
+//   city: 'MOCK_CITY',
+//   postalCode: 'MOCK_PC',
+//   country: 'GB',
+//   cardholderName: 'MOCK_NAME',
+//   cardNumber: '0000000000000000',
+//   expiryDate: '1225',
+//   securityCode: '123',
+//   cardIdentifier: 'FOO',
+//   email: 'mock@example.com',
+// }
```
Comment (bashleigh) [inline packages/payments-client/src/tests/__mocks__/payment.ts:74]: can be deleted?

File: packages/payments-portal/src/components/payment/index.tsx [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    setPaymentProvider: Dispatch<SetStateAction<PaymentProvider | null>>,
+    config: ClientConfigModel | null,
+    paymentModel: PaymentModel | null,
+    propertyModel: PropertyModel | null,
+    merchantKey: MerchantKey | null,
+    receiptAction: ReceiptAction,
+    statusAction: StatusAction,
+    transactionSubmit: (transaction: CreateTransactionModel) => Promise<Transaction>,
+    refreshPayment: () => void,
+  ) =>
```
Comment (bashleigh) [inline packages/payments-portal/src/components/payment/index.tsx:36]: Named parameters please!

File: packages/payments-service/src/payments/controller.ts [hunk 3]
```diff
@@ -1,24 +1,28 @@
...
 
-  @Post('/:paymentId')
-  async getPayment(@Headers() paymentsHeaders: PaymentsHeaders, @Param() paymentId: string): Promise<void> {
-    this.paymentsProvider.getPayment(paymentsHeaders, paymentId)
+  @Get('/:paymentId')
+  async getPayment(
+    @Headers() paymentsHeaders: PaymentsHeaders,
+    @Param() { paymentId }: PaymentsParams,
+  ): Promise<PaymentWithPropertyModel> {
+    return await this.paymentsProvider.getPayment(paymentsHeaders, paymentId)
```
Comment (bashleigh) [inline packages/payments-service/src/payments/controller.ts:17]:
```suggestion
    return this.paymentsProvider.getPayment(paymentsHeaders, paymentId)
```

Don't need the await here

File: packages/payments-service/src/payments/controller.ts [hunk 4]
```diff
@@ -1,24 +1,28 @@
...
   @Patch('/:paymentId')
   async patchPayment(
     @Headers() paymentsHeaders: PaymentsHeaders,
     @Body() paymentPatch: PaymentsDto,
-    @Param() paymentId: string,
-  ): Promise<void> {
-    this.paymentsProvider.patchPayment(paymentsHeaders, paymentPatch, paymentId)
+    @Param() { paymentId }: PaymentsParams,
+  ): Promise<Object> {
+    return await this.paymentsProvider.patchPayment(paymentsHeaders, paymentPatch, paymentId)
```
Comment (bashleigh) [inline packages/payments-service/src/payments/controller.ts:26]:
```suggestion
    return this.paymentsProvider.patchPayment(paymentsHeaders, paymentPatch, paymentId)
```

File: packages/payments-ui/src/payment-state/use-payment-state.tsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
+// import { PaymentModel, PropertyModel } from '@reapit/foundations-ts-definitions'
```
Comment (bashleigh) [inline packages/payments-ui/src/payment-state/use-payment-state.tsx:1]: still need this file?

File: packages/payments-portal/serverless.yml [hunk 6]
```diff
@@ -0,0 +1,105 @@
+service: cloud-payments-portal
```
Comment (bashleigh) [inline packages/payments-portal/serverless.yml:1]: It is possibly we could deploy this with CDK instead of serverless? Mitigates converting in the future

**Review by bashleigh — COMMENTED** [review]
Looks good, only some minor changes, suggestions and question in here.

File: packages/payments-client/src/components/payment/index.tsx [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+      setPaymentProvider,
+      config,
+      payment,
+      property,
+      merchantKey,
+      receiptUpdate,
+      statusUpdate,
+      transactionSubmit,
+      refreshPayment,
+    ),
```
[the remark this answers is not in the document]
Comment (willmcvay) [reply packages/payments-client/src/components/payment/index.tsx:158]: As above

File: packages/payments-client/src/components/payment/index.tsx [hunk 8]
```diff
@@ -0,0 +1,204 @@
...
+          problem persists.
+        </PersistentNotification>
+      </PageContainer>
+    )
+  }
+
+  return <PaymentPageContent paymentProvider={paymentProvider} />
+}
+
+export default PaymentPage
```
[the remark this answers is not in the document]
Comment (willmcvay) [reply packages/payments-client/src/components/payment/index.tsx:204]: It is, yeah, it's loaded by the router

File: packages/payments-client/src/components/payments/payment-request-modal.tsx [hunk 9]
```diff
@@ -31,100 +30,87 @@ export interface PaymentEmailRequestModel {
...
     setSelectedPayment: Dispatch<SetStateAction<PaymentModel | null>>,
     closeModal: () => void,
-    setLoading: Dispatch<SetStateAction<boolean>>,
-    errorSnack: (message: string) => void,
+    updatePayment: SendFunction<UpdateStatusBody, boolean>,
+    generateSession: SendFunction<CreateSessionRequest, boolean | SessionResponse>,
+    generatePaymentRequest: SendFunction<PaymentEmailRequest, boolean>,
     refreshPayments?: () => void,
     clientCode?: string | null,
   ) =>
```
[the remark this answers is not in the document]
Comment (willmcvay) [reply packages/payments-client/src/components/payments/payment-request-modal.tsx:58]: As above, have refactored, the provider constructor params are now a named interface

File: packages/payments-portal/src/components/payment/index.tsx (hunk 2 above)
Comment (willmcvay) [reply packages/payments-portal/src/components/payment/index.tsx:36]: Done

File: packages/payments-service/src/payments/controller.ts (hunk 4 above)
Comment (willmcvay) [reply packages/payments-service/src/payments/controller.ts:26]: Done

File: packages/payments-ui/src/payment-state/use-payment-state.tsx (hunk 5 above)
Comment (willmcvay) [reply packages/payments-ui/src/payment-state/use-payment-state.tsx:1]: Yep, gone

File: packages/payments-client/src/components/payments/payment-request-modal.tsx (hunk 9 above)
[the remark this answers is not in the document]
Comment (willmcvay) [reply packages/payments-client/src/components/payments/payment-request-modal.tsx:58]: Again, not convinced about the readability argument but have updated

File: packages/payments-client/src/components/payments/payment-request-modal.tsx [hunk 10]
```diff
@@ -134,19 +120,57 @@ export const PaymentRequestModal: FC<PaymentRequestModalProps> = ({ refreshPayme
...
           selectedPayment,
           setSelectedPayment,
           closeModal,
-          setLoading,
-          error,
+          updatePayment,
+          generateSession,
+          generatePaymentRequest,
           refreshPayments,
           clientId,
```
[the remark this answers is not in the document]
Comment (willmcvay) [reply packages/payments-client/src/components/payments/payment-request-modal.tsx:175]: Done

File: packages/payments-client/src/tests/__mocks__/payment.ts (hunk 1 above)
Comment (willmcvay) [reply packages/payments-client/src/tests/__mocks__/payment.ts:74]: Done

File: packages/payments-portal/serverless.yml (hunk 6 above)
Comment (willmcvay) [reply packages/payments-portal/serverless.yml]: Done

File: packages/payments-client/src/components/payment/index.tsx [hunk 11]
```diff
@@ -0,0 +1,204 @@
...
+    setPaymentProvider: Dispatch<SetStateAction<PaymentProvider | null>>,
+    config: ClientConfigModel | null,
+    paymentModel: PaymentModel | null,
+    propertyModel: PropertyModel | null,
+    merchantKey: MerchantKey | null,
+    receiptUpdate: ReapitUpdateState<PaymentEmailReceipt, boolean>,
+    statusUpdate: ReapitUpdateState<UpdateStatusBody, boolean>,
+    transactionSubmit: (transaction: CreateTransactionModel) => Promise<Transaction>,
+    refreshPayment: () => void,
+  ) =>
```
[the remark this answers is not in the document]
Comment (bashleigh) [reply packages/payments-client/src/components/payment/index.tsx]: Only more readable when it comes to typehints, a named property will be individually shown whereas index properties are shown in full and hardly to follow. Imo anyway

**Review by bashleigh — APPROVED** [review]

**Pull request merged**